### PR TITLE
/match endpoints

### LINKED ISSUE #157: Create /match endpoints
## Summary
Create 2 new endpoints, `POST /match` and `POST /match/fhir` that behave just like the existing `/link` and `/link/fhir` endpoints without persistence.

## Acceptance Criteria
- [ ] Create two new API endpoints; `/match` and `/match/fhir`
- [ ] Test cases for the new API endpoints
- [ ] Update external documentation explaining the differences between /link and /match

## Background / Context
In some cases, it's optimal for end users to just check to see what person cluster a record belongs to without actually updating the MPI with the new data.  For this action, we'll use the "match" verb to indicate we're just looking for a result.

### PR DESCRIPTION
**Pull request by ericbuckley**
## Description
Add /match and /match/fhir endpoints for link like functionality, however no data is persisted.

## Related Issues
closes #157 

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: src/recordlinker/main.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ async def health_check(
         )
 
 
+app.include_router(link_router, tags=["link"])
```
Comment (cbrinson-rise8) [inline src/recordlinker/main.py:79]: do you no longer need `prefix="/link"`?

Comment (cbrinson-rise8) [reply src/recordlinker/main.py:79]: Never mind I see you manually add the prefix you want in `link_router.py`

**Review by cbrinson-rise8 — APPROVED** [review]
